ParseNOTAM.py

```diff
@@ -12,15 +12,27 @@ def ParseNOTAM(apiOutput=None):
             apiOutput = json.load(json_file)
 
     NOTAMs = []
-    for notam in apiOutput:
-        # Check if 'items' key exists in the notam
-        if 'items' in notam:
-            for item in notam['items']:
-                # Further checks can be added here to ensure the structure of 'item' is as expected
-                if 'properties' in item and 'coreNOTAMData' in item['properties'] and 'notam' in item['properties']['coreNOTAMData']:
-                    NOTAMs.append(Models.Notam(item['properties']['coreNOTAMData']['notam']))
-        else:
-            # Handle the case where 'items' key is missing
-            print(f"Warning: 'items' key missing in notam: {notam}")
+    try:
+        for notam in apiOutput:
+            # Check if 'items' key exists in the notam
+            if 'items' in notam:
+                for item in notam['items']:
+                    try:
+                        core_notam_data = item.get('properties', {}).get('coreNOTAMData', {}).get('notam')
+                        if core_notam_data:
+                            NOTAMs.append(Models.Notam(core_notam_data))
+                        else:
+                            print("Warning: notam missing coreNOTAMData")
+                    except Exception as e:
+                        print("Error processing item: {e}")

```

Comment:
the string value of an Exception in Python is actually not very useful, it only includes the "message" of the Exception, not the Exception type nor the stack trace showing where it occurred.  e.g., if the Exception is an IndexOutOfBoundsException the "message" will be the index number that was out of bounds, like "3", and nothing more, so this `print` here would literally become "Error processing item: 3" (actually here you wouldn't even get that because your string isn't an f-string!) which tells you almost nothing, and in fact is likely to be very misleading making you think the 4th NOTAM is the one with a problem when it fact it could be any NOTAM that failed to process because something tried to access the 4th item in a list or tuple and there wasn't a 4th!

I much prefer to do something like the following for all Exception handling in Python:
```python
except Exception as e:
    import traceback
    print(f"Error processing item: {type(e)} {str(e)}\n{traceback.format_exc(e)}")
```